### DISCUSSION TITLE
Instructions to use the docker and make wheels without Travis

### DISCUSSION THREAD

**Discussion by AbdealiLoKo** [pypa/manylinux#1802] [General]
I saw https://github.com/pypa/python-manylinux-demo which is neat.
But it would also be nice to have instructions that I can simply run on my computer to create the wheels.

Something like:

Step 1: Install docker from the instructions at https://docs.docker.com/engine/installation/
Step 2: Pull the docker image: `docker pull quay.io/pypa/manylinux1_x86_64` (For 64bit)
And so on ...

The Travis build example at https://github.com/pypa/python-manylinux-demo isn't as helpful in case I want to do it locally.

**Comment by techtonik** [comment]
From what I can see:

1. Pull docker image for your architecture

        docker pull quay.io/pypa/manylinux1_x86_64
        # or docker pull quay.io/pypa/manylinux1_i686

**Comment by techtonik** [comment]
2. Run it

    docker run -it -v $(pwd):/io quay.io/pypa/manylinux1_x86_64

3. Then in your `wheelhouse/` directory there will be wheel

**Comment by riddell-stan** [comment]
Here's my version, expanding on @techtonik's a bit:

```bash
docker run -it -v $(pwd):/io quay.io/pypa/manylinux1_x86_64
# following, roughly, https://github.com/pypa/python-manylinux-demo/blob/master/travis/build-wheels.sh
PYBIN=/opt/python/cp36-cp36m/bin
for whl in wheelhouse/*.whl; do
    auditwheel repair "$whl" -w /io/wheelhouse/
done
cp wheelhouse/*.whl /io/wheelhouse/
```

Lots of wheels, including the desired one, will be in ``$(pwd)/wheelhouse``.

**Comment by gfairchild** [comment]
I've followed the instructions so far:

```
docker pull quay.io/pypa/manylinux1_x86_64
docker run -it -v $(pwd):/io quay.io/pypa/manylinux1_x86_64
```

But I don't see a `wheelhouse` directory that you guys seem to have when I'm in the container. Is there something I need to do first to create the directory?

**Comment by riddell-stan** [comment]
Creating a wheelhouse directory in your working directory might do the 
trick.

**Comment by gfairchild** [comment]
But the script you posted above expects `*.whl` files to already exist in `/wheelhouse` when logging into the Docker container:

```bash
for whl in wheelhouse/*.whl; do
    auditwheel repair "$whl" -w /io/wheelhouse/
done
```

How do those files get created?

**Comment by gfairchild** [comment]
Ok, I think I figured it out. From https://github.com/pypa/python-manylinux-demo/blob/master/travis/build-wheels.sh, this is key:

```
for PYBIN in /opt/python/*/bin; do
    "${PYBIN}/pip" wheel /io/ -w wheelhouse/
done
```

**Comment by cdietschrun** [comment]
This process is still unclear to me. I have a repository of Python code I need to build, I'm doing a 
`docker run -it -v $(pwd):/io quay.io/pypa/manylinux1_x86_64`

but I don't see anything inside /io. What am I missing?

**Comment by matthew-brett** [comment]
I think you'll find that starts up the container only for it close immediately.  I think you want something like this:

```
docker run --rm -it -v $(pwd):/io quay.io/pypa/manylinux1_x86_64 /bin/bash
```

This will drop you into the container, in the bash shell, and you will be able to see `/io`.

**Comment by cdietschrun** [comment]
I've run that and other flavors, and io is still empty.

**Comment by cdietschrun** [comment]
Both on x86_64 and i686 versions of the container

**Comment by cdietschrun** [comment]
To be explicit, I'm running on Powershell in Windows and have tried with `pwd`:/io, C:\Users\me\source\repo:/io, /c/Users/me/source/repo etc and all aren't populating io for me

**Comment by cdietschrun** [comment]
Looks like it's just some Windows issue on my setup with -v

**Comment by matthew-brett** [comment]
Ah - sorry - I haven't tried to use Linux containers on Windows.
